docs/conflicts.rst

```diff
@@ -0,0 +1,131 @@
+Namely, updating which Snapshot object a particular (file) name points to (including adding a new name, or removing one).
+
+We do not know *when* another Participant has updated their notion of the current state.
+However, what we *can* notice is when they "commit" to that state (that is, they update publically their current Snapshot pointer).
+
+Each file is considered individually, and is represented by a tree of Snapshots of the contents (plus metadata).
+Included in the metadata are one or more "parent" Snapshots (a brand new file has zero parents).
+
+For more on this, the :ref:`datamodel` has a section on :ref:`datamodel_conflicts`.
+
+
+Communication Between Folders
+-----------------------------
+
+Consider a single file named ``"foo"``.
+When this file is created (for example on the device called Laptop) that device uploads a Snapshot (``S0``) with no parent Snapshots.
+
+By "upload" we mean to push the content and metadata to the Tahoe-LAFS client, receiving ultimately a Capability for the Snapshot ``S0``.
+The file entry for ``"foo"`` is then updated, visualized as: ``foo -> S0``
+Here, "updated" meanst that we use Tahoe-LAFS to edit to contents of our Personal directory to change "foo" to point at Snapshot ``S0``.
+
+Now, all other Participants can (and, eventually, will) notice this update when they poll the magic-folder.
+
+Each of these updates to our Personal directory is a "communication event", so we talk about the regions between these as a cohesive state of that client.
+That is, anything that happens during that time is "at the same time" for the purposes of this protocol.
+
+Note that this time interval could be as short as a few seconds or as long as days (or more).
+
+Observe too that the "parents" of a particular Snapshot are a commitment to the state visible by a particular client when creating any new Snapshot.
+
+
+Detecting Conflicts
+-------------------
+
+A new update is communicated to us -- that is, we've downloaded a previously-unknown Snapshot from some other Participant's mutable Capbility.
+
+We now look at our own Snapshot for the corresponding file.
+Either our Snapshot appears in some parents (including grandparents, etc) of the new Snapshot, or it doesn't.
+
+If it does appear, this is an "update" (and we simply replace the local content with the incoming content).
+
+Instead if our Snapshot does not appear as any ancestor of the incoming Snapshot, a conflict is determined.
+
+This is because when the other device created their Snapshot, they didn't know about ours (or else it would appear as an ancestor) so we have made a change "at the same time".
+
+Note that unlike tools like Git, we do not examine the contents of the file or try to produce differences -- everything is determined from the metadata in the Snapshot.
+This means that even if we happened to make the very same edits "at the same time" it would still be a conflict.
```

Comment:
Curious: Is this true because, presumably, the capability strings would differ (on the assumption that the parties aren't sharing the convergence secret)? What if they are sharing the a convergence secret (and somehow manage to push at the same time)?

(I'm mostly just wondering whether this behavior -- of "it would still be a conflict" -- is intended or accidental here).

Reply:
Doesn't matter if the convergent secret is shared or not, the Snapshot is a new immutable and will be different. Yes, it is intended; if a client makes a new Snapshot, it is a new version whether or not the actual content is identical.